### PR TITLE
feat: [FX-4053] move rich text editor to a separate package

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-4053]

### Description

We have to move RTE to a separate package to provide customers more flexibility

### How to test

- use alpha version in Staff Portal

Use this [PR](https://github.com/toptal/staff-portal/pull/10538) to test

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Ensure that deployed demo has expected results and good examples
- [x] Self reviewed

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4053]: https://toptal-core.atlassian.net/browse/FX-4053?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmaklygin** [comment]
@toptal-bot run package:alpha-release

File: .changeset/tender-poems-editor.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+'@toptal/picasso-rich-text-editor': patch
+'@toptal/picasso-forms': patch
```
Comment (dmaklygin) [inline .changeset/tender-poems-editor.md:3]: wdyt, forms should also be major?

**Comment by dmaklygin** [comment]
@toptal-bot run package:alpha-release

**Comment by dmaklygin** [comment]
@toptal-bot run package:alpha-release

File: .changeset/tender-poems-editor.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+'@toptal/picasso-rich-text-editor': patch
+'@toptal/picasso-forms': major
+---
+
+### RichTextEditor
+
+- create `Rich Text Editor` package. `@toptal/picasso-forms` now expects `@toptal/picasso-rich-text-editor` to be installed separately.
```
Comment (TomasSlama) [inline .changeset/tender-poems-editor.md:8]: I asked ChatGPT to help generating changelog
```suggestion
- @toptal/picasso-rich-text-editor is now considered a peer dependency for this package. This implies that users will need to install it manually. This change provides more control over the versioning of dependencies.
```

**Review by TomasSlama — APPROVED** [review]
Looks good!

**Review by augustobmoura — APPROVED** [review]
It looks like everything is working, didn't catch any possible errors

File: .changeset/tender-poems-beg.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+'@toptal/picasso': major
+---
+
+### RichTextEditor
+
+- remove Rich Text Editor from picasso package. Now, this is a separate package: @toptal/picasso-rich-text-editor
```
[the remark this answers is not in the document]
Comment (dmaklygin) [reply .changeset/tender-poems-beg.md:7]: the PR with a codemod is in a separate PR. I decided to create another PR :)

File: .changeset/tender-poems-editor.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+'@toptal/picasso-rich-text-editor': patch
```
[the remark this answers is not in the document]
Comment (dmaklygin) [reply .changeset/tender-poems-editor.md:2]: yeah, good point

**Pull request merged**